### PR TITLE
FEATURE: Add unit tests for server/restapi and server/config parsing

### PR DESCRIPTION
**Pull request by caohhung**
server/restapi cover:
- next_run_iso_format
- standardized_api_list
- print_plugin
- print_plugins
- get_plugin_history

server/daemon cover:
- _load_config
- _yaml_include
- _yaml_include_dir

### CODE REVIEW

**Comment by pbenas** [comment]
One test fails if there's no `/etc/smokerd/smokerd.yaml`. We can't expect this file being present on a developer's machine.
```
$ py.test
=========================================================== test session starts ============================================================
platform darwin -- Python 2.7.10, pytest-2.9.1, py-1.4.30, pluggy-0.3.1
rootdir: /Users/petr/gdc/smoker, inifile:
collected 75 items / 1 errors

tests/server/test_plugins.py ...........................................................................

================================================================== ERRORS ==================================================================
______________________________________________ ERROR collecting tests/server/test_restapi.py _______________________________________________
tests/server/test_restapi.py:15: in <module>
    class TestRestAPI(object):
tests/server/test_restapi.py:53: in TestRestAPI
    smokerd = Smokerd()
smoker/server/daemon.py:49: in __init__
    self._load_config()
smoker/server/daemon.py:96: in _load_config
    with open(self.conf['config'], 'r') as fp:
E   IOError: [Errno 2] No such file or directory: '/etc/smokerd/smokerd.yaml'
=================================================== 75 passed, 1 error in 27.52 seconds ===================================================
```

**Comment by caohhung** [comment]
I've just updated PR. Using custom smokerd.yaml instead of default /etc/smokerd/smokerd.yaml

**Comment by caohhung** [comment]
btw, I've just added unittest for server/config parsing to this PR.

File: tests/server/test_daemon.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            }
+        }
+    }
+
+    def test_load_config(self):
+        yaml_file = self.conf_dir + '/smokerd.yaml'
+
+        expected_plugins = copy.deepcopy(self.expected_plugins)
+        expected = dict(expected_plugins, **copy.deepcopy(self.expected_basic))
+        # I'm not sure if we still keep 'config' key/value
```
Comment (caohhung) [inline tests/server/test_daemon.py]: Can you please confirm this? It'll remove this comment later

Comment (pbenas) [inline tests/server/test_daemon.py]: looks ok

**Comment by pbenas** [comment]
```
=================================================================== FAILURES ====================================================================
____________________________________________ TestRestAPI.test_forced_print_plugins_with_forced_result ____________________________________________

self = <tests.server.test_restapi.TestRestAPI object at 0x10ba68c50>

    def test_forced_print_plugins_with_forced_result(self):
        plugins_to_print = self.conf_plugins.keys()
        for plugin in self.smokerd.pluginmgr.get_plugins().values():
            plugin.forced = True
            plugin.run()
        time.sleep(0.5)
        plugins_result = restserver.print_plugins(plugins_to_print,
                                                  forced=True)
        assert plugins_result['plugins']
        assert len(plugins_result['plugins']['items']) == len(plugins_to_print)

        for plugin_name in plugins_to_print:
            index = plugins_to_print.index(plugin_name)
            plugin_result = plugins_result['plugins']['items'][index]['plugin']
            assert plugin_result['name'] == plugin_name
            assert plugin_result['links']['self'] == '/plugins/' + plugin_name
            assert plugin_result['parameters']
>           assert plugin_result['forcedResult']['status'] == 'OK'
E           assert 'ERROR' == 'OK'
E             - ERROR
E             + OK
```
My guess is the root cause is `free` command not available on mac... 
```
$ free -m | grep Mem
-bash: free: command not found
```
Please try to come op with `top` command based solution working both on Linux and Mac or get rid of the free mem plugin completely in tests.

**Comment by caohhung** [comment]
Changed from `free` to `uptime` . `top` doesn't solve the problem

**Pull request merged**